pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -192,19 +192,8 @@ public void doRemoveExpiredPrimaryKeys() {
     AtomicInteger numTotalKeysMarkForDeletion = new AtomicInteger();
     AtomicInteger numDeletedKeysWithinTTLWindow = new AtomicInteger();
     double largestSeenComparisonValue = _largestSeenComparisonValue.get();
-    double metadataTTLKeysThreshold;
-    if (_metadataTTL > 0) {
-      metadataTTLKeysThreshold = largestSeenComparisonValue - _metadataTTL;
-    } else {
-      metadataTTLKeysThreshold = Double.MIN_VALUE;
-    }
-    double deletedKeysThreshold;
-    if (_deletedKeysTTL > 0) {
-      deletedKeysThreshold = largestSeenComparisonValue - _deletedKeysTTL;
-    } else {
-      deletedKeysThreshold = Double.MIN_VALUE;
-    }
-
+    double metadataTTLKeysThreshold = _metadataTTL > 0 ? largestSeenComparisonValue - _metadataTTL : 0;
+    double deletedKeysThreshold = _deletedKeysTTL > 0 ? largestSeenComparisonValue - _deletedKeysTTL : 0;
```

Comment:
```suggestion
    double metadataTTLKeysThreshold = _metadataTTL > 0 ? largestSeenComparisonValue - _metadataTTL : LARGEST_COMPARISON_VALUE_NOT_SET;
    double deletedKeysThreshold = _deletedKeysTTL > 0 ? largestSeenComparisonValue - _deletedKeysTTL : LARGEST_COMPARISON_VALUE_NOT_SET;
```

Can change to NEGATIVE_INFINITY for better code consistency

Reply:
as this is to assign value to threshold, it's a bit confusing to reuse this constant value here. 
the removal logic below doesn't use the threshold if metadata/deleted ttl is not enabled, so I set 0 for brevity here.

Reply:
Yeah I saw we check `_metadataTTL / deletedKeysTTL > 0` before doing actual removal. IMO it might be cleaner and safer to set to `NEGATIVE_INFINITY` just in case someone removes the `_metadataTTL / deletedKeysTTL > 0` check in future without context.

Reply:
sgtm, set to NEGATIVE_INFINITY